system/CLI/CLI.php

```diff
@@ -301,6 +301,96 @@ public static function promptByKey($text, array $options, $validation = null): s
         return static::prompt(PHP_EOL . array_shift($text), array_keys($options), $validation);
     }
 
+    /**
+     * * This method is the same as promptByKey(), but this method supports multiple keys, separated by commas.
+     *
+     * @param string $text    Output "field" text or an one or two value array where the first value is the text before listing the options
+     *                        and the second value the text before asking to select one option. Provide empty string to omit
+     * @param array  $options A list of options (array(key => description)), the first option will be the default value
+     *
+     * @return array The selected key and value of $options
+     */
+    public static function promptByMultipleKeys($text, array $options)
+    {
+        if (is_string($text)) {
+            $text = [$text];
+        } elseif (! is_array($text)) {
+            throw new InvalidArgumentException('$text can only be string');
+        }
+
+        if (is_array($options) && $options) {
+            $opts               = $options;
+            $extraOutputDefault = static::color('0', 'green');
+
+            unset($opts[0]);
+
+            if (empty($opts)) {
+                $extraOutput = $extraOutputDefault;
+            } else {
+                $optsKey = [];
+
+                foreach (array_keys($opts) as $key) {
+                    $optsKey[] = $key;
+                }
+                $extraOutput = '[' . $extraOutputDefault . ', ' . implode(', ', $optsKey) . ']';
+            }
+
+            $default = 0;
+        } else {
+            throw new InvalidArgumentException('$options can only be array');
+        }
+
+        if ($line = array_shift($text)) {
+            CLI::newLine();
+            CLI::write($line);
+        }
+
+        // +2 for the square brackets around the key
+        $keyMaxLength = max(array_map('mb_strwidth', array_keys($options))) + 2;
+
+        foreach ($options as $key => $description) {
+            $name = str_pad('  [' . $key . ']  ', $keyMaxLength + 4, ' ');
+            CLI::write(CLI::color($name, 'green') . CLI::wrap($description, 125, $keyMaxLength + 4));
+        }
+        static::fwrite(STDOUT, (trim((string) ((int) $text)) ? ' ' : '') . $extraOutput . ': ');
+        $input = trim(static::input()) ?: $default;
```

Comment:
This will hang for 6 hours in the GH actions because the unit test is waiting for input. If you can in a way extract the non-testable part(s) in another method so that it can be mocked in tests.

Reply:
Now we cannot write test for it. See #6076
It is a problem.

Reply:
Hmm. This is the problem with using static methods. It is not mockable by PHPUnit.

Reply:
@paulbalandan It seems difficult to ask the contributor to add test code.
Do we accept this PR without test code?

Reply:
Ah yes. What I did before in `promptByKey` PR is to check locally the changes then suggest changes, if any, since `prompt` is not testable.